curriculum/challenges/english/20-upcoming-python/learn-tree-traversal-by-building-a-binary-search-tree/655a151cc6041f0ff7d24ded.md

```diff
@@ -0,0 +1,56 @@
+---
+id: 655a151cc6041f0ff7d24ded
+title: Step 15
+challengeType: 20
+dashedName: step-15
+---
+
+# --description--
+
+Now we would recursively traverse the tree and insert the values using the principle for binary trees:
+
+- Values smaller than the key are placed in the left sub-tree
+- Values greater than the key are placed in the right sub-tree
+
+Create a condition that checks if the `key` < `node.key`.
+
+# --hints--
+
+Use an `if` condition for logic.
+    
+```js
+({ test: () => assert.match(code, /if/) });
+```
+
+Check if the `key` < `node.key`.
+
+```js
+({ test: () => assert.match(code, /key < node\.key/) });
+```
```

Comment:
I would replace these two with:

You should create another `if` conditional statement to check if `key < node.key`.

```js
({ test: () => assert.match(e.code.original["main.py"], /^(\s+)if\s+node\s+is\s+None\s*:.*?\1if\s+key\s*<\s*node\.key\s*:/ms) })
```